File: BoardGames/Player.cpp
#include "Player.hpp"
#include <ostream>

Player::Player(const std::string& name) : name(name), winCount(0) {
}


void Player::AddWin() {
	winCount += 1;
}

void Player::SetStats(int wins) {
	winCount = wins;
}


int Player::GetStats() {
	return winCount;
}

std::string Player::GetName() {
	return name;
}


std::ostream& operator<<(std::ostream& os, const Player& p) {
	os << p.name << " (" << p.winCount << " wins)";
	return os;
}

bool operator==(const Player& a, const Player& b) {
	return a.name == b.name;
}

bool operator<(const Player& a, const Player& b) {
	return a.winCount < b.winCount;
}

File: BoardGamesSharp/Core/Player.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BoardGamesSharp.Core;

public class Player
{
    private string name;
    private int winCount;

    public Player(string in_name)
    {
        name = in_name;
        winCount = 0;
    }

    public void AddWin() {
        winCount++;
    }

    public int Wins
    {
        get { return winCount; }
        set { winCount = value; }
    }   

    public string Name
    {
        get { return name; }
    }
}
